cardillo/math/rotations.py

```diff
@@ -554,12 +554,11 @@ def Exp_SO3_quat(P, normalize=True):
     Nuetzi2016: https://www.research-collection.ethz.ch/handle/20.500.11850/117165
     """
     p0, p = np.array_split(P, [1])
-    p_tilde = ax2skew(p)
     if normalize:
         P2 = P @ P
-        return eye3 + (2 / P2) * (p0 * p_tilde + p_tilde @ p_tilde)
+        return eye3 + (2 / P2) * (p0 * ax2skew(p) + ax2skew_squared(p))
     else:
-        return eye3 + 2 * (p0 * p_tilde + p_tilde @ p_tilde)
```

Comment:
I would suggest to use another implementation for normalize==False. 
The current implementation returns neither an orthogonal nor an normalized matrix for non-unit quaternions. 
Equation 6.163 from Egeland returns orthogonal matrices even for non-unit quaternions
`return (p0**2 - p @ p) * eye3 + np.outer(2*p, p) + ax2skew(2*p0*p)`

Reply:
Are equations 6.163 and 6.166 not the same?

Reply:
Only for unit quaternions. I will have a look at the testing and the (possible) changes in Exp_SO3_quat and derived functions as soon as this pull request is merged.